scenarios/aws/microVMs/microvms/run.go

```diff
@@ -320,11 +336,59 @@ func run(e commonConfig.CommonEnvironment) (*ScenarioDone, error) {
 		return nil, err
 	}
 
-	microVMIPMap := GetDomainIPMap(vmCollections)
-	for domainID, ip := range microVMIPMap {
-		instanceEnv.Ctx.Export(domainID, pulumi.String(ip))
+	// populate microVM IP mapping
+	for _, collection := range vmCollections {
+		for _, domain := range collection.domains {
+			instanceEnv.Ctx.Export(domain.domainID, pulumi.String(domain.ip))
+		}
 	}
 
+	// provision microVMs
+	for _, collection := range vmCollections {
+		if collection.instance.Arch == LocalVMSet {
+			continue
+		}
+
+		sshConfigDone, err := setupMicroVMSSHConfig(collection.instance, microVMGroupSubnet, waitFor)
+		if err != nil {
+			return nil, err
+		}
+
+		microVMSSHKey, readKeyDone, err := readMicroVMSSHKey(collection.instance, sshConfigDone)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, domain := range collection.domains {
+			if domain.lvDomain == nil {
+				continue
+			}
+
+			pc := createProxyConnection(pulumi.String(domain.ip), "root", microVMSSHKey, collection.instance.Connection)
+			remoteRunner, err := command.NewRunner(
```

Comment:
If I understand correctly, you are relying on the retry mechanism to repeat until the micro-vms network us up?